ts/main.ts

```diff
@@ -48,17 +52,44 @@ $entryForm.addEventListener('submit', (event: Event) => {
   const title = $titleInput.value;
   const photoUrl = $photoUrlInput.value;
   const notes = $notesTextArea.value;
-  const entryValues: Entry = {
-    title,
-    photoUrl,
-    notes,
-    entryId: data.nextEntryId,
-  };
-  data.nextEntryId++;
-  data.entries.unshift(entryValues);
-  writeData();
-  const $newEntry = renderEntry(entryValues);
-  $allEntriesUl.prepend($newEntry);
+  if (data.editing === null) {
+    const entryValues: Entry = {
+      title,
+      photoUrl,
+      notes,
+      entryId: data.nextEntryId,
+    };
+    data.nextEntryId++;
+    data.entries.unshift(entryValues);
+    writeData();
+    const $newEntry = renderEntry(entryValues);
+    $allEntriesUl.prepend($newEntry);
+  } else {
+    const entryValues: Entry = {
+      title,
+      photoUrl,
+      notes,
+      entryId: data.editing.entryId,
+    };
+    let editEntryIndex = -1;
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        editEntryIndex = i;
+        break;
+      }
+    }
+    data.entries[editEntryIndex] = entryValues;
```

Comment:
Do we need to have a base `editEntryIndex` here? If we got to the `editing` page, we should know that this item exists in our `data.entries`. You can just have your line 81 inside of your `for` loop's `if` statement and remove the `editEntryIndex` variable completely.